File: src/test/java/lightningtalk/ConcreteEventSource.java
package lightningtalk;

public class ConcreteEventSource implements EventSource {
	@Override
	public void registerListener(EventListener eventListener) {
		eventListener.onEvent();
	}

	public static void main(String[] args) {
		EventSource source = new ConcreteEventSource();
		new SimpleEventListener(source);
	}
}

File: src/test/java/lightningtalk/scala/ConcreteEventSource.scala
package lightningtalk

class ConcreteEventSource extends EventSource{
  override def registerListener(eventListener:EventListener) = eventListener.onEvent  

}

object ConcreteEventSource {
  def main(args:Array[String]) = {
    val source = new ConcreteEventSource
    new SimpleEventListener(source)
  }
}